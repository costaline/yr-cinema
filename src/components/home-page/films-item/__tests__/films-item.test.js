import React from 'react';
import { shallow } from 'enzyme';

import FilmsItem from '../films-item';

describe('Film item', () => {
  const props = {
    name: null,
    posterURL: null
  };

  describe('all data', () => {
    const nextProps = {
      ...props,
      name: 'Some name',
      posterURL: 'https://site.com/image.jpg'
    };

    const wrapper = shallow(<FilmsItem {...nextProps} />);

    it('render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('correct property', () => {
      expect(wrapper.find('h3').text()).toEqual(nextProps.name);
      expect(wrapper.find('img').prop('src')).toEqual(nextProps.posterURL);
    });
  });
});

// TODO: add more tests
