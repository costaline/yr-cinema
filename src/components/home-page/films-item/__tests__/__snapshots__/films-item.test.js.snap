// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film item all data render correctly 1`] = `
<styled.div>
  <div>
    <h3>
      Some name
    </h3>
    <Loader />
    <img
      alt="Some name"
      onError={[Function]}
      onLoad={[Function]}
      src="https://site.com/image.jpg"
    />
  </div>
</styled.div>
`;
